doc/specifications/proposed/named_queries/developer_documentation.md

```diff
@@ -0,0 +1,180 @@
+# Listing Content and Locations
+
+Starting from Platform 2015.07, listing content and locations in a project is
+much easier, using a couple standardized mechanisms.
+
+Predefined, named queries can be defined in any bundle. They can then be used from twig, using a
+[custom controller](#query_controller), or from PHP code using the [QueryType Registry](#fixme). The controller supports
+[template overriding](#template_override), making it easy to use different templates for different calls.
+
+## <a name="query_controller"></a> The query controller
+A new controller is added : `ez_query`.
+
+It has two actions: `locationQuery`, and `contentQuery`.
+
+Both requires two arguments:
+- `queryName`, the name of the QueryType to run. Example: `AcmeBundle:LatestArticles'
+- `viewType`, similar to what `viewContent` and `viewLocation` expect. Examples: `list`, `tree`, `full_list`...
+
+If the QueryType supports any, they can be provided using the `parameters` argument (hash):
+
+```twig
+{{ render( controller(
+    'ez_query:locationQuery',
+    {
+        'queryName': 'AcmeBundle:LatestContent',
+        'viewType': 'list',
+        parameters: { ContentType: 'article' }
+)) }}
+```
+
+### <a name="template_override"></a>Rendering of queries
+The `ez_query` actions support template override. They support several matchers.
+- `QueryType\Name`: matches the QueryType's name
+- `QueryType\Parameter`: matches the value of one or several of the parameters passed to the QueryType
+- `QueryType\Expression`: matches the parameters with an [Expression](http://symfony.com/en/doc/current/components/expression_language/index.html).
+
+Assuming we want to render the `AcmeBundle:LatestContent` query using
+`AcmeBundle:query/list/latest_articles.html.twig` when the `ContentType` parameter has the value 'article':
+
+```twig
+system:
+    default:
+        query:
+            list:
+                latest_articles:
+                    template: "AcmeBundle:query/list/latest_articles.html.twig"
+                    match:
+                        QueryType\Name: 'AcmeBundle:LatestContent'
+                        QueryType\Parameters: {ContentType: "article"}
+```
+
+#### Template matchers
+
+##### `QueryType\Name`
+Matches the QueryType's name (exact match).
+
+##### `QueryType\Parameters`
+Matches the parameters against a given hash. The hash must contain the parameter's name as the key,
+and the match value as the value:
+
+```twig
+match:
+  QueryType\Parameters:
+    type: "article",
+    category: "development"
+
+Each parameter will be matched exactly.
+The matcher will match if ALL of the provided parameters match what is contained in the hash.
```

Comment:
Yep. We weren't really sure how to distinguish expressions from pure scalars. Overall, it could be added later if it really appears to be missing, and the "global" expression matcher can do everything that would be done this way.